### ISSUE TITLE
`NewType` cannot be combined with `Annotated`

### ISSUE THREAD

**Issue by bluenote10** [microsoft/pyright#10682]
**Describe the bug**

Pyright doesn't seem to allow combining a `NewType` with an `Annotated`. Mypy supports that, and on first glance there is no (obvious) reason why this shouldn't be allowed.

**Code or Screenshots**

```python
from typing import Annotated, NewType

MyType = NewType("MyType", Annotated[int, "whatever"])
```

This errors with ([pyright playground](https://pyright-play.net/?code=FAMwTg9gtgBALgTwA4EsB2BzGKpImOGAQTTQjgEM4BTAEwBoYA5agdwBVlrhgBZBTkmowAvMzaDqACgBE-STMYkylGrQDa6OIxmsAFlWoA3amBkBdAJTAgA)):

```
Expected class as second argument to NewType  (reportGeneralTypeIssues)
```

For comparison: [mypy playground](https://mypy-play.net/?mypy=latest&python=3.12&gist=c16bb5a13abede152d1b01b851be5b8c)

On first glance, I'd expect the type checker to treat `NewType[..., Annotated[int, ...]]` exactly as `NewType[..., int]`, which would indicate that it should type check.

Possibly related to #7383

**Comment by erictraut** [comment]
I don't think this should be allowed. The typing spec is quite clear that the second argument to `NewType` must be a "base class". The expression `Annotated[int, "whatever"]` is not a class. See [this section of the spec](https://typing.python.org/en/latest/spec/aliases.html#newtype) for details.

If you'd like to suggest changes to the typing spec, [here](https://github.com/python/typing-council#decisions) is the process.

**Comment by bluenote10** [comment]
But shouldn't `Annotated[int, ...]` be recognized as base class? The specs of [`Annotated`](https://typing.python.org/en/latest/spec/qualifiers.html#meaning) state:

> Thus, in general, any [type expression](https://typing.python.org/en/latest/spec/glossary.html#term-type-expression) or [annotation expression](https://typing.python.org/en/latest/spec/glossary.html#term-annotation-expression) may be wrapped in Annotated without changing the meaning of the wrapped expression.

So considering that `int` is a valid base class, `Annotated[int, ...]` should be accepted as a valid base class as well.

This also seems to be the case in general, no? The following type checks fine in both mypy and pyright, and should be valid at runtime:

```py
from typing import Annotated

T = Annotated[int, "whatever"]

class SubT(T): # proof that T is a valid base class?
    ...
    
x = SubT(42)
assert isinstance(x, int)
assert isinstance(x, SubT)
```

Given the "proof" that `T` is a valid base class, I'd expect `MyType = NewType("MyType", T)` to be valid according to the specs.

**Comment by bluenote10** [comment]
@erictraut Would it make sense to re-open this issue based on this reasoning?

**Comment by erictraut** [comment]
I don't fully buy your reasoning. The type of the expression `Annotated[int, "whatever"]` at runtime is not a class object. It's an instance of `typing.__AnnotatedAlias`. It acts like a class in some cases but not in others.

Why do you want to use an `Annotated` instance in a `NewType`? I presume that it's not for static type checking, since `Annotated` metatdata is generally ignored by static type checkers. `NewType` is primarily a mechanism for static type checking rather than for runtime applications, so the combination of these two is not something I'd expect to see.

If this is something you'd really like to see supported, then I recommend that you pursue a clarification in the Python typing spec. A clarification like this shouldn't require a full PEP, but it will require public discussion. See [here](https://github.com/python/typing-council#decisions) for the documented process.

**Comment by bluenote10** [comment]
I don't really understand where the unclarity is. The specs say it must be a "base class", and this relatively clearly proves that it is a "base class":

<img width="346" height="110" alt="Image" src="https://github.com/user-attachments/assets/91e4beb3-1c78-40d1-b6e8-2c0842dfdc24" />

After all, we are using it as "base class", no?

My use case was to come up with a pydantic validate-annoated type that is type narrowing. In my opinion the standard usages of `Annotated` in pydantic are not particularly great, because people believe it's a newtype. I often see something like this:

```py
PositiveInt = Annotated[int, postive_int_validator]

def f(x: PositiveInt):
    # code assumes `x` satisfies the invariant that the type name suggests

x: int = 42
f(x) # type checks fine; invariant violated
```

Subclassing is a solution, but has drawbacks (in particular with `str`, because pydantic handles str subclasses inconsistent). Combining `NewType` with `Annotated` is interesting, because it at least prevents implicit conversion (but admittedly doesn't _guarantee_ validation out-of-the-box). That's why I though it might be nice to be able to define it as:

```py
PositiveInt = NewType("PositiveInt", Annotated[int, postive_int_validator])
```

In a sense `Annotated[..., validator]` is a _runtime narrowing_, and `NewType` is a _static narrowing_, so it makes some sense to see them in combination.

I noticed that it's possible to swap `Annotated` with `NewType`, but it requires to introduce another helper type, so I have a work around.

Nonetheless I don't understand the interpretation of the specs, and I think mypy is correct in supporting that.
